packages/flutter_tools/lib/src/device.dart

```diff
@@ -915,6 +915,43 @@ class DebuggingOptions {
 
   bool get hasObservatoryPort => hostVmServicePort != null;
 
+  List<String> getIOSLaunchArguments(EnvironmentType environmentType, String? route,  Map<String, Object?> platformArgs) {
+    final String dartVmFlags = computeDartVmFlags(this);
+    return <String>[
+      '--enable-dart-profiling',
+      if (environmentType == EnvironmentType.physical) '--disable-service-auth-codes',
+      if (environmentType == EnvironmentType.simulator && disableServiceAuthCodes) '--disable-service-auth-codes',
+      if (environmentType == EnvironmentType.physical && disablePortPublication) '--disable-observatory-publication',
+      if (startPaused) '--start-paused',
+      if (environmentType == EnvironmentType.physical && dartVmFlags.isNotEmpty) '--dart-flags="$dartVmFlags"',
```

Comment:
With quotes: [#33924](https://github.com/flutter/flutter/pull/33924/files#diff-5a3d215e39f292ac706d495f042ef45fab9f13962022f4e8e4580254e8557d52R288)
And without quotes: [#63416](https://github.com/flutter/flutter/pull/63416/files#diff-fbf1a86439c67e75e90a00f32fd2b789323582ced2f4f5cfea2d27a1af18aa30R427)

I know you were going to work on https://github.com/flutter/flutter/issues/114622 next, so I guess leave this to be resolved in the next PR.

Reply:
This could use a comment though to explain why they are different.

Reply:
Nit: bump on putting in a comment, even if it's something like "Wrap dart-flags passed to the device in quotes"?